### PR TITLE
[proxy] Implement per-endpoint traffic metrics

### PR DESCRIPTION
**Pull request by funbringer**
Part of https://github.com/neondatabase/cloud/issues/2715

### CODE REVIEW

File: proxy/src/proxy.rs [hunk 1]
```diff
@@ -38,6 +38,15 @@ static NUM_BYTES_PROXIED_COUNTER: Lazy<IntCounter> = Lazy::new(|| {
     .unwrap()
 });
 
+static NUM_BYTES_PROXIED_PER_CLIENT_COUNTER: Lazy<IntCounterVec> = Lazy::new(|| {
```
Comment (vadim2404) [inline proxy/src/proxy.rs:41]: @funbringer , what do you think of having two different numbers for income and outcome traffic? Why am I asking? AWS income traffic costs 0 and these numbers can help us in the future to understand how can we make our solutions cheaper or more expensive based on these numbers.

https://aws.amazon.com/en/ec2/pricing/on-demand/

Comment (funbringer) [reply proxy/src/proxy.rs:41]: Good idea!

**Comment by funbringer** [comment]
We should first merge https://github.com/neondatabase/cloud/pull/2974.

**Review by kelvich — APPROVED** [review]

File: proxy/src/proxy.rs [hunk 2]
```diff
@@ -30,10 +30,11 @@ static NUM_CONNECTIONS_CLOSED_COUNTER: Lazy<IntCounter> = Lazy::new(|| {
...
 
-static NUM_BYTES_PROXIED_COUNTER: Lazy<IntCounter> = Lazy::new(|| {
-    register_int_counter!(
-        "proxy_io_bytes_total",
-        "Number of bytes sent/received between any client and backend."
+static NUM_BYTES_PROXIED_COUNTER: Lazy<IntCounterVec> = Lazy::new(|| {
+    register_int_counter_vec!(
+        "proxy_io_bytes_per_client",
+        "Number of bytes sent/received between client and backend.",
+        &["direction", "tenant_id"]
```
[the remark this answers is not in the document]
Comment (funbringer) [reply proxy/src/proxy.rs:37]: Right, forgot to change it, thanks! I'll use `endpoint_id` then.

**Pull request merged**